lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -892,7 +903,7 @@ macro_rules! stats_sampler {
         }
 
         impl Sampler for $name {
-            fn label(&self) -> &str {
+            fn label(&self) -> &'static str {
```

Comment:
Using `'static` str might cause problems later if we will need dynamically-created samplers (for example, for user-defined visualizations). Am I right that it's only needed because of the usage of `CowString` in `MetricsAccumulator`?

Reply:
Yes; I guess I had reservations against allocating a new String on every call to `push(LabeledSamples)` 😅  and I wasn't sure how to do it otherwise; but maybe I shouldn't worry about that? 🤔 I'm also kind of still uneasy about passing all the LabeledMetrics as just a vector without guarantees of position & labels match; currently I want to add a check verifying the length & labels on every `push()`; but I'm also wondering if I should refactor `Panel` usage in `Monitor` to try and keep it in some tuple or something? 🤔 Though keeping a vector seems somewhat lighter conceptually and maybe also easier to tweak in the future? 🤔 

I now think I will maybe try to experiment a bit more with a slightly different approach to managing the strings: to try and keep a single global `Vector<String>` in `frame_stats`, managed by `push()`; maybe I could then try passing borrowed strings to `push()` for comparison, store them in the `Vector` just the first time, and then the intervals would keep references to that `Vector`. Not 100% sure if my Rust-fu will be enough to achieve that, but I will and want to try. Anyway, if you'd have any suggestions this way or another on how you think I should try improving this, I'd be super grateful and would love to hear them! ❤️

Reply:
After the discussion we had on Discord, I switched to `ImString`. Thanks for the help! ❤️